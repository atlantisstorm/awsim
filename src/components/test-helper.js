import React from 'react';
//import { render } from "@testing-library/react";
import Context from "../store/context";
import { initialiseReducer } from "../store/reducer";

export const RenderWithProvider = (props) => {
  const { children, testState, testDispatch }  = props;
  // Note: we pass any supplied testState to initialiseReducer rather than switching it like
  // we do for dispatch because of scoping.  Basically each rerender after a click/select
  // update event would mean we refresh again with the original data.
  const { state, dispatch: iDispatch } = initialiseReducer({ initialState: testState });
  const dispatch = testDispatch ? testDispatch : iDispatch;
  return (
    <Context.Provider value={{ state, dispatch }}>
      { children }
    </Context.Provider> 
  )
}