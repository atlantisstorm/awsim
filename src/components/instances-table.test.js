import React from 'react';
import { render } from "@testing-library/react";
import InstanceTable from './instances-table';
import { RenderWithProvider } from './test-helper';

describe('InstanceTable component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <RenderWithProvider>
        <InstanceTable />
      </RenderWithProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});