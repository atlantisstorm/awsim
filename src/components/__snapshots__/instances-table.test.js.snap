// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceTable component should match snapshot 1`] = `
<DocumentFragment>
  <table
    class="roundedTable"
  >
    <thead>
      <tr>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Name 
            </div>
            <div
              class="right"
            >
              â–²
            </div>
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Environment 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              State Name 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Public Ip Address 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Private Ip Address 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Auto Scaling Group Name 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Platform 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Instance Id 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Image Id 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Instance Type 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
        <th
          class="{fieldName} {extra_class}"
          data-testid="header-sort"
        >
          <div
            class="title"
          >
            <div
              class="left"
            >
              Launch Time 
            </div>
            <div
              class="right"
            />
          </div>
        </th>
      </tr>
    </thead>
    <tbody />
    <tfoot>
      <tr>
        <td
          colspan="12"
        >
          0 instances found.
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;
